### PR TITLE
[DEV APPROVED] [9508] Implement a Decorator class that adds helpers pagination methods

### PR DESCRIPTION
**Pull request by MarcoBgn**
@tomas-stefano following our discussion this is one possible solution for extracting the decorator/presenter.

### CODE REVIEW

File: spec/site_search/decorator_spec.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      it { expect(subject.next_page).to eq(1) }
+    end
+
+    context 'when there is not a next page' do
+      let(:page) { 1 }
+
+      it { expect(subject.next_page).to eq(nil) }
+    end
+  end
+end
```
Comment (tomas-stefano) [inline spec/site_search/decorator_spec.rb]: What do you think to change to:

`expect(subject.first_page?).to be(true)` to `expect(subject).to be_first_page`
`expect(subject.last_page?).to be(true)` to `expect(subject).to be_last_page`

File: spec/site_search/decorator_spec.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    context 'when it is not the last page' do
+      it { expect(subject.last_page?).to eq(false) }
+    end
+  end
+
+  describe '#previous_page' do
+    context 'when there is a previous page' do
+      let(:page) { 1 }
+
+      it { expect(subject.previous_page).to eq(0) }
```
Comment (tomas-stefano) [inline spec/site_search/decorator_spec.rb]: What do you think to rename to: `expect(subject.previous_page).to be_zero`

File: spec/site_search/decorator_spec.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    end
+
+    context 'when there is not a previous page' do
+      it { expect(subject.previous_page).to eq(nil) }
```
Comment (tomas-stefano) [inline spec/site_search/decorator_spec.rb]: What do you think to rename to `expect(subject.previous_page).to be_nil`

File: spec/site_search/decorator_spec.rb (hunk 1 above)
Comment (MarcoBgn) [reply spec/site_search/decorator_spec.rb]: 👍

File: spec/site_search/decorator_spec.rb (hunk 2 above)
Comment (MarcoBgn) [reply spec/site_search/decorator_spec.rb]: Yep, good point

File: spec/site_search/decorator_spec.rb (hunk 3 above)
Comment (MarcoBgn) [reply spec/site_search/decorator_spec.rb]: 👍

**Comment by MarcoBgn** [comment]
Thanks @tomas-stefano, comments addressed

**Pull request merged**